src/SSHDebugPS/UI/ContainerPickerDialogWindow.xaml

```diff
@@ -292,154 +404,157 @@
-                  Grid.Row="2"
-                  Grid.ColumnSpan="2"
-                  Margin="0,10,0,0"
-                  VerticalAlignment="Top"
-                  ExpandDirection="Down">
-            <Grid>
-                <Grid.RowDefinitions>
-                    <RowDefinition Height="Auto" />
-                </Grid.RowDefinitions>
-                <Grid.ColumnDefinitions>
-                    <ColumnDefinition Width="Auto" />
-                    <ColumnDefinition Width="*" />
-                </Grid.ColumnDefinitions>
-                
-                <Label x:Uid="uidHostnameLabel"
-                       Grid.Row="0"
-                       Grid.Column="0"
-                       x:Name="Hostname"
-                       Target="{Binding ElementName=hostnameTextBox}"
-                       Margin="19,5,0,5"
-                       Content="{x:Static local:UIResources.HostnameLabel}"
-                       ToolTip="{x:Static local:UIResources.HostnameTip}" />
+        <!-- Hostname label and textbox -->
 
-                <TextBox x:Uid="uidHostnameTextBox"
-                         Grid.Row="0"
-                         Grid.Column="1"
-                         x:Name="hostnameTextBox"
-                         Text="{Binding Path=Hostname, Mode=TwoWay,UpdateSourceTrigger=PropertyChanged}"
-                         PreviewKeyDown="HostnameTextbox_PreviewKeyDown"
-                         Margin="0,5,0,5"
-                         Padding="2" />
-                </Grid>
-        </Expander>
+        <Label Grid.Row="2"
+               Grid.ColumnSpan="2"
+               x:Uid="uidHostnameLabel"
+               x:Name="HostnameLabel"
+               Margin="0,9,0,0"
+               Padding="0"
+               Height="16"
+               VerticalAlignment="Center"
+               Target="{Binding ElementName=hostnameTextBox}"
+               Content="{x:Static local:UIResources.HostnameLabel}" />
+        <TextBox Grid.Row="3"
+                 Grid.ColumnSpan="2"
+                 Margin="0,4,0,0"
+                 Padding="2"
+                 x:Uid="uidHostnameTextBox"
+                 x:Name="hostnameTextBox"
+                 AutomationProperties.Name="DockerHostnameOptional"
+                 PreviewKeyDown="HostnameTextbox_PreviewKeyDown"
+                 Text="{Binding Path=Hostname, Mode=TwoWay,UpdateSourceTrigger=PropertyChanged}"
+                 ToolTip="{x:Static local:UIResources.HostnameTip}">
+            <TextBox.Style>
+                <Style TargetType="TextBox">
+                    <Style.Triggers>
+                        <DataTrigger Binding="{Binding ElementName=hostnameTextBox, Path=Text.IsEmpty}"
+                                     Value="True">
+                            <Setter Property="Text"
+                                    Value="{x:Static local:UIResources.OptionalHintText}" />
+                        </DataTrigger>
```

Comment:
This looks like a watermark? In that case should the text color get lighter?

Reply:
Oops. That ended up not working so i removed it and added (Optional) to the label